src/librustc_mir/borrow_check/nll/region_infer/error_reporting/mod.rs

```diff
@@ -253,28 +253,40 @@ impl<'tcx> RegionInferenceContext<'tcx> {
             let outgoing_edges_from_graph = self.constraint_graph
                 .outgoing_edges(r, &self.constraints, fr_static);
 
-
-            // But member constraints can also give rise to `'r: 'x`
-            // edges that were not part of the graph initially, so
-            // watch out for those.
-            let outgoing_edges_from_picks = self.applied_member_constraints(r)
-                .iter()
-                .map(|&AppliedMemberConstraint { min_choice, member_constraint_index, .. }| {
-                    let p_c = &self.member_constraints[member_constraint_index];
-                    OutlivesConstraint {
-                        sup: r,
-                        sub: min_choice,
-                        locations: Locations::All(p_c.definition_span),
-                        category: ConstraintCategory::OpaqueType,
+            macro_rules! handle_constraint {
```

Comment:
I would have used a closure but 🤷‍♀️

Reply:
I did it this way because I wasn't sure if you could guarantee that a closure gets inline. But then I learned it's possible with `#![feature(stmt_expr_attributes)]`. So I'll use that.